api/create.go

```diff
@@ -79,26 +80,37 @@ func CreateService(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	authHeader := req.Header.Get("Authorization")
-	userId, err := config.ValidateToken(authHeader)
+	apiKeyHeader := req.Header.Get("x-api-key")
+
+	userId, err := config.ValidateUser(authHeader, apiKeyHeader)
 	if err != nil {
-		log.Printf("error validating token %v", err)
-		http.Error(w, "error validating token", http.StatusUnauthorized)
+		log.Printf(err.Error())
+		http.Error(w, "error validating user", http.StatusUnauthorized)
 		return
 	}
+
 	var s service
+
 	byteArray, err := ioutil.ReadAll(req.Body)
 	if err != nil {
 		log.Fatalf("fatal: reading from readall body %v", err)
 	}
+
 	err = json.Unmarshal(byteArray, &s)
+	if s.UserID == "" && apiKeyHeader != "" {
+		s.UserID = "testuser"
+	}
+
```

Comment:
I guess we can remove this if we go with the suggestion above - we don't have to do another extra check for `apiKeyHeader` since `ValidateUser()` already does that.